ttter/accounts/views.py

```diff
@@ -47,119 +49,104 @@ def form_invalid(self, form):
 
 
 class UserListView(LoginRequiredMixin, ListView):
-    template_name = 'accounts/user_list.html'
+    template_name = "accounts/user_list.html"
     model = User
-    
+
     def get_queryset(self):
         return User.objects.all()
 
 
 class UserPage(LoginRequiredMixin, TemplateView):
     template_name = "accounts/user_page.html"
+
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         request_user = self.request.user
-        print(self.kwargs ,1111111111111111111) 
-        user = get_object_or_404(User, id=self.kwargs.get("user_id"))
+        user = get_object_or_404(User, username=self.kwargs.get("username"))
         post_item = Tweet.objects.filter(user=user)
-        context["is_followed"] = FriendShip.objects.filter(follower=request_user, followee=user).exists()
-        print(FriendShip.objects.filter(followee=request_user, follower=user).exists())
+        context["is_followed"] = FriendShip.objects.filter(
+            followee=request_user, follower=user
+        ).exists()
         context["request_user"] = request_user
         context["user"] = user
         context["post_item"] = post_item
+        context["followee_count"] = user.followee.count()
+        context["follower_count"] = user.follower.count()
         return context
 
-    # def get_queryset(self, **kwargs):
-    #     context = super().get_queryset(**kwargs)
-    #     print(type(self.request.user.id))
-    #     print(context)
-    #     return User.objects.get(id=int(context['username']))
-
-    # def get_queryset(self):
-    #     return User.objects.get(id=self.request.user.id)
-
 
 class FollowerListView(LoginRequiredMixin, TemplateView):
     template_name = "accounts/follower_list.html"
 
     def get_context_data(self, **kwargs):
         ctx = super().get_context_data(**kwargs)
-        print(self.kwargs ,1111111111111111111)
-        print(FriendShip.objects.filter(follower_id=self.kwargs.get("user_id")))
-        ctx["followers"] = FriendShip.objects.filter(follower_id=self.kwargs.get("user_id"))
+        self.user = get_object_or_404(User, username=self.kwargs["username"])
+
+        ctx["followers"] = self.user.follower.all()
```

Comment:
こちらもall以外のやり方でリストを取得

Reply:
こちら、修正してみたのですが、うまくいきません
```suggestion
        ctx["followings"] = [follower for follower in self.user.all().prefetch_related("followers")]
```

Reply:
FriendShipオブジェクトからfollowerデータを抽出してみてください。

Reply:
なるほどです！
ありがとうございます